src/Spaces/finitedifference.jl

```diff
@@ -1,25 +1,35 @@
 abstract type AbstractFiniteDifferenceSpace <: AbstractSpace end
 
-abstract type Staggering end
+"""
+    FiniteDifferenceSpace(
+        grid::Grids.FiniteDifferenceGrid,
+        staggering::Staggering, 
+    )
 
-""" Cell center location """
-struct CellCenter <: Staggering end
 
-""" Cell face location """
-struct CellFace <: Staggering end
 
+"""
 struct FiniteDifferenceSpace{
+    G <: Grids.AbstractFiniteDifferenceGrid,
     S <: Staggering,
-    T <: Topologies.AbstractIntervalTopology,
-    GG,
-    LG,
 } <: AbstractFiniteDifferenceSpace
+    grid::G
     staggering::S
-    topology::T
-    global_geometry::GG
-    center_local_geometry::LG
-    face_local_geometry::LG
 end
+FiniteDifferenceSpace(
+    topology::Topologies.IntervalTopology,
+    staggering::Staggering,
+) = FiniteDifferenceSpace(Grids.FiniteDifferenceGrid(topology), staggering)
+
+
+const FaceFiniteDifferenceSpace{G} = FiniteDifferenceSpace{G, CellFace}
+const CenterFiniteDifferenceSpace{G} = FiniteDifferenceSpace{G, CellCenter}
+
+grid(space::AbstractFiniteDifferenceSpace) = getfield(space, :grid)
+staggering(space::FiniteDifferenceSpace) = getfield(space, :staggering)
```

Comment:
Any particular reason why we're using `getfield` here over `getproperty`?

Reply:
`getproperty` was causing JET errors, because it hit the deprecation. I don't quite understand why